File: yixiang_fe/src/types/notification.ts
export type NotificationType = 'LIKE' | 'COMMENT' | 'FOLLOW' | 'SYSTEM';

export interface NotificationItem {
  id: number;
  actorId: number | null;
  actorNickname: string | null;
  actorAvatar: string | null;
  type: NotificationType;
  entityType: 'POST' | 'COMMENT' | null;
  entityId: number | null;
  content: string | null;
  isRead: boolean;
  createdAt: string;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  nextCursor: number | null;
  hasMore: boolean;
}

export interface UnreadCountResponse {
  unreadCount: number;
}

File: yixiang_be/src/main/java/com/tongji/notification/model/Notification.java
package com.tongji.notification.model;

import lombok.Data;
import java.time.Instant;

@Data
public class Notification {
    private Long id;
    private Long recipientId;
    private Long actorId;
    private String type;
    private String entityType;
    private Long entityId;
    private String content;
    private boolean isRead;
    private Instant createdAt;
}
